src/state_machine.zig

```diff
@@ -1399,7 +1409,7 @@ fn check(comptime test_table: []const u8) !void {
                     1,
                     commit_operation,
```

Comment:
I see we do an alloc every `.commit` here. Is it intentional? I am _guessing_ that this is indeed so to take advantage of `@memset(buffer, undefined)` in the allocator impl, but I am wondering if maybe I am overthinking this, and we just didn't bother to pull this alloc out of the loop?

Reply:
It is the latter; I'm not too careful w/ allocations in tests. :stuck_out_tongue: